sys/include/net/gnrc/mac/types.h

```diff
@@ -46,6 +49,68 @@ typedef enum {
     TX_FEEDBACK_BUSY          /* found medium busy when doing transmission */
 } gnrc_mac_tx_feedback_t;
 
+/**
+ * @brief MAC internal type for storing reception state parameters and
+ *        state machines.
+ *        This structure can be extended to contain more needed
+ *        states and parameters. Please #ifdef+NEWMAC them out for particular
+ *        new MAC protocols.
+ */
+typedef struct {
+    /* RX queue for storing received packets */
+    gnrc_priority_pktqueue_t queue;
+
+    /* RX queue nodes */
+    gnrc_priority_pktqueue_node_t _queue_nodes[GNRC_MAC_RX_QUEUE_SIZE];
+
+    /* buffer for storing dispatching packets */
+    gnrc_pktsnip_t* dispatch_buffer[GNRC_MAC_DISPATCH_BUFFER_SIZE];
+} gnrc_mac_rx_t;
+
+/**
+ * @brief type for storing states of tx neighbor node.
+ */
+typedef struct {
+    uint8_t  l2_addr[IEEE802154_LONG_ADDRESS_LEN]; /* Address of neighbour node */
+    uint8_t  l2_addr_len;                          /* Neighbour address length */
+    gnrc_priority_pktqueue_t queue;                /* TX queue for this particular Neighbour */
+    uint32_t phase;                                /* Neighbour's wake-up Phase */
+} gnrc_mac_tx_neighbour_t;
+
+/**
+ * @brief Uninitialized phase value.
+ */
+#define GNRC_MAC_PHASE_UNINITIALIZED   (0)
+
+/**
+ * @brief Maximum phase value.
+ */
+#define GNRC_MAC_PHASE_MAX             (-1)
+
+/**
+ * @brief MAC internal type for storing transmission state parameters and
+ *        state machines.
+ *        This structure can be extended to contain more needed
+ *        states and parameters. Please #ifdef+NEWMAC them out for particular
+ *        new MAC protocols.
+ */
+typedef struct {
+    /* TX queues for neighbouring nodes. First queue is for broadcast (+1) */
+    gnrc_mac_tx_neighbour_t neighbours[GNRC_MAC_NEIGHBOUR_COUNT + 1];
```

Comment:
> Maybe I need rework on this.. (how about setting GNRC_MAC_NEIGHBOUR_COUNT to 0, if the protocol doesn't need this.)

then you need to put a

```C
#if GNRC_MAC_NEIGHBOUR_COUNT != 0
    /* ... */
#endif
```

around the array. Otherwise the compiler won't be too happy ;-)